pages/checkout/index.js

```diff
@@ -143,6 +147,24 @@ class CheckoutPage extends Component {
     })
   }
 
+  handleDiscountChange = e => {
+    e.preventDefault();
+    if (this.state.discountCode.trim() && this.props.checkout) {
```

Comment:
```suggestion
    if (!this.state.discountCode.trim() || !this.props.checkout) {
      return;
    }
    ...
```

You can reduce nesting this way